lldb/source/Plugins/Process/gdb-remote/GDBRemoteCommunicationClient.cpp

```diff
@@ -403,6 +403,22 @@ void GDBRemoteCommunicationClient::GetRemoteQSupported() {
         x.split(compressions, ',');
         if (!compressions.empty())
           MaybeEnableCompression(compressions);
+      } else if (x.consume_front("SupportedWatchpointTypes=")) {
+        llvm::SmallVector<llvm::StringRef, 4> watchpoint_types;
+        x.split(watchpoint_types, ',');
+        m_watchpoint_types =
+            WatchpointHardwareFeature::eWatchpointHardwareFeatureUnknown;
+        for (auto wp_type : watchpoint_types) {
+          if (wp_type == "x86_64")
+            m_watchpoint_types |=
+                WatchpointHardwareFeature::eWatchpointHardwareX86;
+          if (wp_type == "aarch64-mask")
+            m_watchpoint_types |=
+                WatchpointHardwareFeature::eWatchpointHardwareArmMASK;
+          if (wp_type == "aarch64-bas")
+            m_watchpoint_types |=
+                WatchpointHardwareFeature::eWatchpointHardwareArmBAS;
+        }
```

Comment:
Please put this string -> enum code next to the enum if possible. Perhaps `eWatchpointHardwareFeatureUnknown` could be the fallback value, assuming that that is 0.

Then you can loop over the strings here doing:
m_watchpoint_types |= stringToType(..);

And we'll have one place to update/reference in future.

Reply:
Also I wonder if this should be case insensitive, but I doubt it'll ever be an issue. If compression types does that then sure, if not, it's fine as is.

Reply:
I thought about putting the gdb remote serial protocol names in lldb-enumerations.h but they're gdb RSP specific strings and I don't expect them to be used anywhere else.

Reply:
Sure. The other motivation for me was to not add more else if to `GetRemoteQSupported` which is probably a gazillion lines long. But a few more on top of that isn't changing anything so it's fine.